ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -2871,10 +2871,19 @@ public boolean skipProcessing(Task<?> task) {
       Path loadPath = dpCtx.getRootPath();
       FileSystem fs = loadPath.getFileSystem(conf);
       int numDPCols = dpCtx.getNumDPCols();
-      Map<Path, Optional<List<Path>>> allPartition = new HashMap<>();
+      Map<Path, Optional<List<FileStatus>>> allPartition = new HashMap<>();
       if (dynamicPartitionSpecs != null) {
         for (Map.Entry<String, List<Path>> partSpec : dynamicPartitionSpecs.entrySet()) {
-          allPartition.put(new Path(loadPath, partSpec.getKey()), Optional.of(partSpec.getValue()));
+          if (Optional.of(partSpec.getValue()).isPresent()) {
```

Comment:
Would this break HIVE-24682 ?

Reply:
I think it's same. I have just changed the data structure to have FileStatus instead of Path. If path is already null (isPresent is false) ..i am adding FileStatus as null.